src/main/kotlin/io/github/mojira/arisa/modules/RemoveVersionModule.kt

```diff
@@ -0,0 +1,38 @@
+package io.github.mojira.arisa.modules
+
+import arrow.core.Either
+import arrow.core.extensions.fx
+import io.github.mojira.arisa.domain.Issue
+import java.time.Instant
+
+class RemoveVersionModule : Module {
+    override fun invoke(issue: Issue, lastRun: Instant): Either<ModuleError, ModuleResponse> = with(issue) {
+        Either.fx {
+            val addedVersions = getExtraVersionsLatelyAddedByNonVolunteers(lastRun)
+            val removeAddedVersions = affectedVersions
+                .filter { it.id in addedVersions }
+                .map { it.remove }
+            assertNotEmpty(removeAddedVersions).bind()
+            removeAddedVersions.forEach(::run)
+        }
+    }
+
+    private fun Issue.getExtraVersionsLatelyAddedByNonVolunteers(lastRun: Instant): List<String> =
+        if (created.isAfter(lastRun)) {
+            if (isVolunteer(reporter?.getGroups?.invoke())) {
+                emptyList()
+            } else {
+                affectedVersions.map { ver -> ver.id }
+            }
+        } else {
+            changeLog
+                .asSequence()
+                .filter { it.created.isAfter(lastRun) }

```

Comment:
I think it would be better additionally to this, to also check for after the issue's resolved date. If Arisa has a longer downtime, there could be a case where a version is added, and the issue is resolved afterwards.